### PR TITLE
Refactor docblock annotation parser

### PR DESCRIPTION
**Pull request by smoench**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -43,19 +43,20 @@
...
         "php": "^7.3",
         "ext-phar": "*",
+        "composer-plugin-api": "^1.0 || ^2.0",
         "amphp/parallel-functions": "^0.1.3",
         "beberlei/assert": "^3.2",
         "composer/package-versions-deprecated": "^1.8",
         "composer/semver": "^3.2",
         "composer/xdebug-handler": "^1.3.2",
-        "composer-plugin-api": "^1.0 || ^2.0",
         "hoa/compiler": "^3.17",
```
Comment (villfa) [inline composer.json:52]: `hoa/compiler` could be removed now, right?

Comment (smoench) [reply composer.json]: removed

**Comment by smoench** [comment]
@theofidry Do you agree with those changes? If yes then I would adjust those failing test cases.

**Comment by theofidry** [comment]
@smoench yes LGTM.

I don't feel strongly about that compactor. Although I tried to lean towards an exhaustive test suite, the goal is to remove as much phpdoc as possible without breaking anything. So keep in mind whilst fixing the tests that if there is a case too complicated to handle we can leave the phpdoc in that case it's fine

**Comment by smoench** [comment]
Tests are passing

**Comment by theofidry** [comment]
👏  nice one

**Pull request merged**